torch/_dynamo/symbolic_convert.py

```diff
@@ -306,7 +306,7 @@ def wrapper(self: "InstructionTranslatorBase", inst: Instruction):
             try:
                 return inner_fn(self, inst)
             except Unsupported as excp:
-                if self.has_backedge():
+                if self.has_backedge() and self.should_compile_partial_graph():
```

Comment:
Probably not related to this PR, but should we unify the exception handling of [step](https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/symbolic_convert.py#L498-L515) and of this `break_graph_if_unsupported` decorator into a single helper to prevent divergence or is it expected to have different implementation?